File: sdks/typescript/src/llm_api_protocol_values.ts
/** HTTP API protocols used by supported LLM endpoints. */
export const LlmApiProtocol = Object.freeze({
  ChatCompletions: "chat_completions",
  Responses: "responses",
  Anthropic: "anthropic",
} as const);

export type LlmApiProtocol = (typeof LlmApiProtocol)[keyof typeof LlmApiProtocol];

export function llmApiProtocolEndpointPath(protocol: LlmApiProtocol): string {
  switch (protocol) {
    case LlmApiProtocol.Responses: return "responses";
    case LlmApiProtocol.Anthropic: return "messages";
    case LlmApiProtocol.ChatCompletions: return "chat/completions";
  }
}

export function llmApiProtocolTryFromEndpoint(endpoint: string): LlmApiProtocol | undefined {
  const base = (endpoint.split(/[?#]/u)[0] ?? endpoint).replace(/\/+$/u, "");
  if (base.endsWith("/responses")) return LlmApiProtocol.Responses;
  if (base.endsWith("/messages")) return LlmApiProtocol.Anthropic;
  if (base.endsWith("/chat/completions")) return LlmApiProtocol.ChatCompletions;
  return undefined;
}

export function llmApiProtocolFromEndpoint(endpoint: string): LlmApiProtocol {
  const detected = llmApiProtocolTryFromEndpoint(endpoint);
  if (detected !== undefined) return detected;
  const base = (endpoint.split(/[?#]/u)[0] ?? endpoint);
  return base.includes("anthropic.com/") ? LlmApiProtocol.Anthropic : LlmApiProtocol.ChatCompletions;
}

File: sdks/python/src/echo_agent_sdk/llm_api_protocol_values.py
from __future__ import annotations

from enum import Enum


class LlmApiProtocol(str, Enum):
    """HTTP API protocol spoken by a complete provider endpoint."""

    CHAT_COMPLETIONS = "chat_completions"
    RESPONSES = "responses"
    ANTHROPIC = "anthropic"

    def endpoint_path(self) -> str:
        if self is LlmApiProtocol.RESPONSES:
            return "responses"
        if self is LlmApiProtocol.ANTHROPIC:
            return "messages"
        return "chat/completions"

    @classmethod
    def try_from_endpoint(cls, endpoint: str) -> LlmApiProtocol | None:
        base = endpoint.split("?", 1)[0].split("#", 1)[0].rstrip("/")
        if base.endswith("/responses"):
            return cls.RESPONSES
        if base.endswith("/messages"):
            return cls.ANTHROPIC
        if base.endswith("/chat/completions"):
            return cls.CHAT_COMPLETIONS
        return None

    @classmethod
    def from_endpoint(cls, endpoint: str) -> LlmApiProtocol:
        detected = cls.try_from_endpoint(endpoint)
        if detected is not None:
            return detected
        base = endpoint.split("?", 1)[0].split("#", 1)[0]
        return cls.ANTHROPIC if "anthropic.com/" in base else cls.CHAT_COMPLETIONS
